packages/common/src/lib/core/utils/utils.ts

```diff
@@ -34,19 +34,19 @@ export function getHttpProxyAgent(proxyUrl?: string): HttpProxyAgent | HttpsProx
     console.log(`The proxy server at '${proxy}' will be used.`);
   }
   
-  if (proxy?.startsWith('https')) {
+  if (downloadUrl?.startsWith('https')) {
     return new HttpsProxyAgent(proxyAgentOpts);
-  } else if(proxy?.startsWith('http')){
+  } else if(downloadUrl?.startsWith('http')){
     return new HttpProxyAgent(proxyAgentOpts);
   } else {
     return undefined
   }
 }
 
-export function getCommonHttpHeaders( pkgName: string, proxyUrl?: string){
+export function getCommonHttpHeaders( pkgName: string, proxyUrl?: string, downloadUrl?: string){
```

Comment:
can you please : 
- make `downloadUrl`  **not optional**,
- rename it into a more generic `targetUrl` 
- and so move it _before_  `proxyUrl`
- 
as such:

```
export function getCommonHttpHeaders( pkgName: string, targetUrl?: string, proxyUrl?: string){
```